graph/src/blockchain/types.rs

```diff
@@ -333,6 +333,21 @@ pub struct ChainIdentifier {
     pub genesis_block_hash: BlockHash,
 }
 
+impl ChainIdentifier {
+    pub fn is_default(&self) -> bool {
+        ChainIdentifier::default().eq(self)
+    }
+}
+
+impl Default for ChainIdentifier {
+    fn default() -> Self {
+        Self {
+            net_version: String::default(),
+            genesis_block_hash: BlockHash::from(H256::zero()),
+        }
+    }
+}
```

Comment:
it doesn't mean anything special, it's just easier to build than repeating the same code everywhere

Reply:
What changes?

Reply:
If you are referring to [these](https://github.com/graphprotocol/graph-node/pull/5337/files/1623b904c455504b58d6fbcdbe89d2ba09b46b9f#diff-5fc766188c2b64f33c6143f3377c76c8d0f9994c5f93e151f60985c973422691R2306) then they are not strictly necessary. I'm not a big fan of implicit behavior in the sense that the current implementations writes some warning to update the net_version if there are some special values (again somewhat implicit except for a comment). 

I thought it would be more correct to update the value to the latest used. If you prefer not to then I'll revert those changes.